xds/server_ext_test.go

```diff
@@ -217,6 +223,242 @@ func (s) TestServingModeChanges(t *testing.T) {
 	waitForFailedRPCWithStatus(ctx, t, cc, errAcceptAndClose)
 }
 
+// TestServingModeChanges_MultipleServers tests two servers with unique
+// bootstrap configuration are able to serve two different clients with same
+// name if they are in different xds client pools.
+//
+// It tests the Server's logic as it transitions from Not Ready to Ready, then
+// then to Not Ready. Before it goes Ready, connections should be accepted and
+// closed. After it goes ready, RPC's should proceed as normal according to
+// matched route configuration. After it transitions back into not ready
+// (through an explicit LDS Resource Not Found), previously running RPC's
+// should be gracefully closed and still work, and new RPC's should fail.
+func (s) TestServingModeChanges_MultipleServers(t *testing.T) {
```

Comment:
I don't think we need to test serving mode changes here. All we need to do here is to ensure that we can create two servers with two different bootstrap configurations. This would also mean that the two servers will request two different LDS resources from the management server, because the LDS resource name depends on the port on which the server is listening.

Then we could have two gRPC clients that talk to each of these servers and we ensure that RPCs work. We need to ensure that a gRPC client is actually talking to the server that it intends to talk to. This can be done with the `Peer` call option.